spot-client/src/common/remote-control/remoteControlServer.js

```diff
@@ -93,6 +93,10 @@ export class RemoteControlServer extends BaseRemoteControlService {
      * @override
      */
     getJoinCode() {
+        if (this._options.backend) {
+            return this._options.joinCode;
```

Comment:
As far as I understand the code, the `_options.joinCode` is the latest join code used in `connect`.

In the backend scenario it never gets stale. Maybe it could read from `_previousJoinCode`, but then it has `previous` in the name and might be confusing.

I was thinking to have those strategy classes and store this code in the backend strategy instance.

Another thing that could be done is to have a dedicated `joinCode` field that would cache the join code after refresh (either from XMPP or backend).